packages/salesforcedx-vscode-core/src/conflict/metadataCacheService.ts

```diff
@@ -252,21 +258,64 @@ export class MetadataCacheService {
   }
 
   /**
-   * Groups the information in a MetadataCacheResult by component
+   * Groups the information in a MetadataCacheResult by component.
+   * Child components are returned as an array entry unless their parent is present.
    * @param result A MetadataCacheResult
    * @returns An array with one entry per retrieved component, with all corresponding information about the component included
    */
   public static correlateResults(result: MetadataCacheResult): CorrelatedComponent[] {
     const components: CorrelatedComponent[] = [];
 
-    const projectIndex = new Map<string, SourceComponent>();
+    const projectIndex = new Map<string, RecomposedComponent>();
     for (const comp of result.project.components) {
-      projectIndex.set(MetadataCacheService.makeKey(comp.type.name, comp.fullName), comp);
+      const key = MetadataCacheService.makeKey(comp.type.name, comp.fullName);
+      if (comp.parent) {
+        const parentKey = MetadataCacheService.makeKey(comp.parent.type.name, comp.parent.fullName);
```

Comment:
The next few lines are quite complex and will help to add why we are doing this. Even better if we can somehow refactor it into a method on its own (and reuse for cacheIndex).

Reply:
I agree, will do that!! I added a bunch of comments in the latest commit, let me know what you think